internal/olm/operator/uninstall.go

```diff
@@ -82,88 +89,116 @@ func (u *Uninstall) Run(ctx context.Context) error {
+	// 3. ClusterServiceVersion. OLM puts an ownerref on every namespaced resource to the CSV,
+	//    and an owner label on every cluster scoped resource so they get gc'd on deletion.
+	// 4. CatalogSource. All other resources installed by OLM or operator-sdk related to this
+	//    package will be gc'd.
+
+	// Subscriptions can be deleted asynchronously.
+	if err := u.deleteObjects(ctx, false, subObj); err != nil {
+		return err
+	}
+	var objs []client.Object
+	if u.DeleteCRDs {
+		objs = append(objs, crds...)
+	}
+	objs = append(objs, csvObj, csObj)
+	// These objects may have owned resources/finalizers, so block on deletion.
+	if err := u.deleteObjects(ctx, true, objs...); err != nil {
 		return err
 	}
 
-	// If this was the last subscription in the namespace and the operator group is
-	// the one we created, delete it
+	// If the last subscription in the namespace was deleted and the operator group is
+	// the one operator-sdk created, delete it.
 	if u.DeleteOperatorGroups {
-		if err := u.config.Client.List(ctx, &subs, client.InNamespace(u.config.Namespace)); err != nil {
-			return fmt.Errorf("list subscriptions: %v", err)
-		}
-		if len(subs.Items) == 0 {
-			ogs := v1.OperatorGroupList{}
-			if err := u.config.Client.List(ctx, &ogs, client.InNamespace(u.config.Namespace)); err != nil {
-				return fmt.Errorf("list operatorgroups: %v", err)
-			}
-			for _, og := range ogs.Items {
-				og := og
-				if len(u.DeleteOperatorGroupNames) == 0 || slice.ContainsString(u.DeleteOperatorGroupNames, og.GetName(), nil) {
-					if err := u.deleteObjects(ctx, false, &og); err != nil {
-						return err
-					}
-				}
-			}
+		if err := u.deleteOperatorGroup(ctx); err != nil {
+			return err
 		}
 	}
-	if sub == nil {
+
+	// If no objects were cleaned up, the package was not found.
+	if subObj == nil && csObj == nil && csvObj == nil && len(crds) == 0 {
 		return &ErrPackageNotFound{u.Package}
 	}
 	return nil
 }
 
+func (u *Uninstall) deleteOperatorGroup(ctx context.Context) error {
+	subs := v1alpha1.SubscriptionList{}
+	if err := u.config.Client.List(ctx, &subs, client.InNamespace(u.config.Namespace)); err != nil {
+		return fmt.Errorf("list subscriptions: %v", err)
+	}
+	if len(subs.Items) != 0 {
+		return nil
+	}
+	ogs := v1.OperatorGroupList{}
+	if err := u.config.Client.List(ctx, &ogs, client.InNamespace(u.config.Namespace)); err != nil {
+		return fmt.Errorf("list operatorgroups: %v", err)
+	}
+	for _, og := range ogs.Items {
+		if len(u.DeleteOperatorGroupNames) == 0 || slice.ContainsString(u.DeleteOperatorGroupNames, og.GetName(), nil) {
+			if err := u.deleteObjects(ctx, false, &og); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
 func (u *Uninstall) deleteObjects(ctx context.Context, waitForDelete bool, objs ...client.Object) error {
 	for _, obj := range objs {
-		obj := obj
-		lowerKind := strings.ToLower(obj.GetObjectKind().GroupVersionKind().Kind)
```

Comment:
Why we are changing the command? Should we not only change the tests? 
Is it a fix? If yes, we need to add the fragment.

Reply:
Looks like this command broke in the last release so this should have a fragment.